pageserver/src/page_service.rs

```diff
@@ -554,37 +554,49 @@ struct BatchedGetPageRequest {
     timer: SmgrOpTimer,
 }
 
+#[cfg(feature = "testing")]
+struct BatchedTestRequest {
+    req: models::PagestreamTestRequest,
+    timer: SmgrOpTimer,
+}
+
 enum BatchedFeMessage {
     Exists {
         span: Span,
         timer: SmgrOpTimer,
-        shard: timeline::handle::Handle<TenantManagerTypes>,
+        shard: timeline::handle::WeakHandle<TenantManagerTypes>,
```

Comment:
nit: it might be nice with a comment on the enum reminding the reader why these hold weak handles and where to read more about it.

Reply:
66c0df81090bb0f93e468b53cec485f2a4605554